velox/type/LongDecimal.h

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <folly/dynamic.h>
+#include <sstream>
+#include <string>
+#include "velox/common/base/BitUtil.h"
+#include "velox/common/base/Exceptions.h"
+#include "velox/type/StringView.h"
+
+#pragma once
+
+#define BUILD_INT128(X, Y) ((int128_t)(X) << 64 | Y)
+
+namespace facebook::velox {
+
+using int128_t = __int128_t;
+
+struct LongDecimal {
+  // Default required for creating vector with NULL values.
+  constexpr explicit LongDecimal() : unscaledValue(0) {}
+  constexpr explicit LongDecimal(int128_t value) : unscaledValue(value) {}
+
+  bool operator==(const LongDecimal& other) const {
+    return unscaledValue == other.unscaledValue;
+  }
+
+  bool operator!=(const LongDecimal& other) const {
+    return unscaledValue != other.unscaledValue;
+  }
+
+  bool operator<(const LongDecimal& other) const {
+    return unscaledValue < other.unscaledValue;
+  }
+
+  bool operator<=(const LongDecimal& other) const {
+    return unscaledValue <= other.unscaledValue;
+  }
+  int128_t unscaledValue;
```

Comment:
ShortDecimal struct has unscaledValue a private member variable and provides a getter. It would be nice to make Long and ShortDecimal structs the same.